src/compiler/clvm.rs

```diff
@@ -608,10 +608,10 @@ pub fn parse_and_run(
     }
 }
 
-fn sha256tree_from_atom(v: Vec<u8>) -> Vec<u8> {
+pub fn sha256tree_from_atom(v: &[u8]) -> Vec<u8> {
     sha256(
         Bytes::new(Some(BytesFromType::Raw(vec![1])))
-            .concat(&Bytes::new(Some(BytesFromType::Raw(v)))),
+            .concat(&Bytes::new(Some(BytesFromType::Raw(v.to_vec())))),
```

Comment:
it looks quite strange to heap-allocate and copy the bytes here, just to concatenate and run sha-256 over them. That should be possible without any heap allocation or copying (and possibly less code).

Reply:
slicked these up and added a test.